src/graphics.rs

```diff
@@ -1628,6 +1634,263 @@ impl Drawable for Sprite {
     }
 }
 
+// AsciiSubset refers to the subset of ascii characters which give alphanumeric characters plus symbols
+pub enum CharacterListType {
+    AsciiSubset,
+    Ascii,
+    Cyrillic,
+    Thai,
+    Vietnamese,
+    Chinese,
+    Japanese,
+}
+
+#[derive(Debug, Clone)]
+struct CharInfo {
+    vertical_offset: i32,
+    width: f32,
+    uvs: Box2<f32>,
+}
+
+/// `FontTexture` is a texture generated using the *_character_list functions.
+/// It contains a texture representing all of the rasterized characters
+/// retrieved from the *_character_list function. `font_map` represents a
+/// a mapping between a character and its respective character texture
+/// located within `font_texture`.
+#[derive(Debug)]
+pub struct FontAtlas {
+    font_map: HashMap<char, CharInfo>,
+    font_texture: Cached<Texture>,
+}
+
+impl FontAtlas {
+    pub fn new<R: Read>(
+        ctx: &mut Graphics,
+        font: R,
+        font_size: f32,
+        char_list_type: CharacterListType,
+    ) -> Result<FontAtlas> {
+        let inval_bb = ::rusttype::Rect {
+            min: ::rusttype::Point { x: 0, y: 0 },
+            max: ::rusttype::Point {
+                x: (font_size / 4.0) as i32,
+                y: 0,
+            },
+        };
+        const MARGIN: usize = 3;
+        let char_list = Self::get_char_list(char_list_type)?;
+        println!("This many chars: {}", char_list.len());
+        let chars_per_row = ((char_list.len() as f32).sqrt() as usize) + 1;
+        let bytes_font = font.bytes().collect::<Result<Vec<u8>, IoError>>()?;
+        let rusttype_font = ::rusttype::Font::try_from_bytes(&bytes_font[..]).ok_or(anyhow!(
```

Comment:
`Font` seems like something to include at the top level. Once this is moved to a separate `text.rs` file, ofc.